.github/workflows/self-test.yml

```diff
@@ -0,0 +1,29 @@
+---
+name: Self-Test
+
+on: # NOLINT
+  pull_request:
+  push:
+  workflow_dispatch:
```

Comment:
should we add a weekly test as well?

Reply:
I don't know, I went back and forth on that.  What do you think?

Reply:
I'm fine without, this is a pretty simple repository.